filters/openpolicyagent/opaauthorizerequest/benchmark_test.go

```diff
@@ -0,0 +1,421 @@
+			ContentLength: int64(len(body)),
+		},
+		FMetrics: &metricstest.MockMetrics{},
+	}
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			f.Request(ctx)
+		}
+	})
+
+}
+
+func BenchmarkJwtValidation(b *testing.B) {
+	publicKey, err := os.ReadFile(certPath)
+	if err != nil {
+		b.Fatalf("Failed to read public key: %v", err)
+	}
+
+	opaControlPlane := opasdktest.MustNewServer(
+		opasdktest.MockBundle(testBundleEndpoint, map[string]string{
+			"main.rego": fmt.Sprintf(`
+					package envoy.authz
+
+					import future.keywords.if
+
+					default allow = false
+
+					public_key_cert := %q
+
+					bearer_token := t if {
+						v := input.attributes.request.http.headers.authorization
+						startswith(v, "Bearer ")
+						t := substring(v, count("Bearer "), -1)
+					}
+
+					allow if {
+						[valid, _, payload] :=  io.jwt.decode_verify(bearer_token, {
+							"cert": public_key_cert,
+							"aud": "nqz3xhorr5"
+						})
+					
+						valid
+						
+						payload.sub == "5974934733"
+					}				
+				`, publicKey),
+		}),
+	)
+
+	filterOpts := NewFilterOptionsWithDefaults(opaControlPlane.URL())
+	f, err := createOpaFilter(filterOpts)
+	require.NoError(b, err)
+
+	reqUrl, err := url.Parse("http://opa-authorized.test/somepath")
+	require.NoError(b, err)
+
+	claims := jwt.MapClaims{
+		"iss":   "https://some.identity.acme.com",
+		"sub":   "5974934733",
+		"aud":   "nqz3xhorr5",
+		"iat":   time.Now().Add(-time.Minute).UTC().Unix(),
+		"exp":   time.Now().Add(tokenExp).UTC().Unix(),
+		"email": "someone@example.org",
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+
+	privKey, err := os.ReadFile(keyPath)
+	require.NoError(b, err, "Failed to read priv key from path %q", keyPath)
+
+	key, err := jwt.ParseRSAPrivateKeyFromPEM(privKey)
+	require.NoError(b, err, "Failed to parse RSA PEM")
+
+	// Sign and get the complete encoded token as a string using the secret
+	signedToken, err := token.SignedString(key)
+	require.NoError(b, err, "Failed to sign token")
+
+	ctx := &filtertest.Context{
+		FStateBag: map[string]interface{}{},
+		FResponse: &http.Response{},
+		FRequest: &http.Request{
+			Header: map[string][]string{
+				"Authorization": {fmt.Sprintf("Bearer %s", signedToken)},
+			},
+			URL: reqUrl,
+		},
+		FMetrics: &metricstest.MockMetrics{},
+	}
+
+	b.ResetTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			f.Request(ctx)
+			assert.False(b, ctx.FServed)
+		}
+	})
+}
+
+// BenchmarkPolicyBundle measures the performance of the OPA authorization filter
+// with a policy bundle loaded from a .tar.gz file.
+
+// This benchmark, serving as a reference test case, evaluates the
+// filter's overhead when using a pre-compiled policy bundle to make authorization decisions.
+
+// To generate a bundle in .tar.gz format, use the following command:
+// ./opa build -b <bundle_directory> -o <output_file.tar.gz>
+// For example:
+// ./opa build -b simple-opa-bundle -o simple-opa-bundle.tar.gz
```

Comment:
You should mention that you have a pre-made directory in this repo. I.e. if you include a line before that goes to that directory it would help